usecases/objects/batch_references_add.go

```diff
@@ -109,6 +113,12 @@ func (b *BatchManager) validateReference(wg *sync.WaitGroup, ref *models.BatchRe
 		err = joinErrors(errors)
 	}
 
+	if shouldValidateMultiTenantRef(tenantKey, source, target) && err == nil {
```

Comment:
I would check the error first to avoid doing unecessary check `if err== nil && shouldValidateMultiTenantRef(tenantKey, sou rce, target)`

Reply:
The condition `shouldValidateMultiTenantRef(tenantKey, source, target) && err == nil` is equivalent to `err == nil && shouldValidateMultiTenantRef(tenantKey, source, target)`. However, the latter is more efficient if err != nil, `shouldValidateMultiTenantRef()` will not be called unnecessarily

Reply:
ah ok, makes sense. i will change it